cmd/kubeadm/app/cmd/alpha/kubelet.go

```diff
@@ -109,16 +111,39 @@ func newCmdKubeletConfigDownload() *cobra.Command {
 			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
 			kubeadmutil.CheckErr(err)
 
-			err = kubeletphase.DownloadConfig(client, kubeletVersion, constants.KubeletRunDirectory)
+			criSocket, err := getCRISocket(criSocketStr)
+			if err != nil {
+				kubeadmutil.CheckErr(err)
+			}
+
+			nodeReg := kubeadmapi.NodeRegistrationOptions{
+				CRISocket: criSocket,
+			}
+
+			err = kubeletphase.DownloadConfig(client, kubeletVersion, &nodeReg, constants.KubeletRunDirectory)
 			kubeadmutil.CheckErr(err)
 		},
 	}
 
 	options.AddKubeConfigFlag(cmd.Flags(), &kubeConfigFile)
+	cmdutil.AddCRISocketFlag(cmd.Flags(), &criSocketStr)
```

Comment:
this mutates the downloaded KubeletConfiguration so that the user can override CRI socket or use autodetection.
this might be the right thing to do.

@rosti PTAL.